### ISSUE TITLE
vim.tigrc: G move-last-line does not work in main view

### ISSUE THREAD

**Issue by kiryph** [jonas/tig#594]
I have configured tig to use to the vim bindings defined in `contrib/vim.tigrc` in `~/tigrc`:
```
source ~/.config/tig/vim.tigrc
```

```
$ tig --version
tig version 2.2.2
```

Why does it not work in the main view but in the diff view?

**Comment by loonies** [comment]
Add `bind main G move-last-line`

**Comment by kiryph** [comment]
Thanks. @jonas Do you mind adding this by default to `contrib/vim.tigrc`?

**Comment by jonas** [comment]
@kiryph The current behaviour is a bug. Do you want to make a PR?

**Comment by kiryph** [comment]
@jonas Thanks for you response. I am not sure if you refer to tig itself or the vim.tigrc configuration file. I would assume if I `bind main G none` and in the next line I set a generic one, I do not need to `bind main G` explicitly.

If you mean that this is unintentional and should be fixed in something like https://github.com/jonas/tig/blob/master/src/keys.c#L155-L183, I am afraid I would have to understand your source code to create a PR without screwing something up. Naively, I would expect something like this

```diff
❯ git diff
diff --git a/src/keys.c b/src/keys.c
index e405320..594227d 100644
--- a/src/keys.c
+++ b/src/keys.c
@@ -177,6 +177,8 @@ get_keybinding(const struct keymap *keymap, const struct key key[], size_t keys,
                        (*matches) += generic_matches;
                if (request == REQ_UNKNOWN)
                        request = generic_request;
+               if (request == REQ_NONE && generic_matches == 1)
+                       request = generic_request;
        }

        return request == REQ_NONE ? REQ_UNKNOWN : request;
```

**Comment by jonas** [comment]
I think it should be fixed in the vim.tigrc file. Your proposed fix will break the ability to unassigned a generic keybinding in a specific keymap. This was one of the motivations for introducing `none` as a key action. For true unassignment it would be better to introduce an `unbind` command.

**Comment by jonas** [comment]
Fixed by #599 in 416072f.
